src/transform/src/cse/relation_cse.rs

```diff
@@ -36,11 +55,13 @@ impl crate::Transform for RelationCSE {
     fn transform(
         &self,
         relation: &mut MirRelationExpr,
-        _: TransformArgs,
+        args: TransformArgs,
     ) -> Result<(), crate::TransformError> {
         let mut bindings = Bindings::default();
         bindings.intern_expression(relation)?;
         bindings.populate_expression(relation);
+        self.inline_let.transform_without_trace(relation)?;
```

Comment:
Hmm, we now have a situation where the need to trace a call is conditional based on the client..

I'm not super fond of having to introduce dedicated methods such as `transform_without_trace` just to disable. I think that we should have it the other way around - `transform` should not trace the call, and instead the `Transform` trait should implement a default `trace_transform` method that adds the necessary decoration around the `self.transform()` call. This will also address [@frankmcsherry's concern  from #16157](https://github.com/MaterializeInc/materialize/pull/16157#discussion_r1028140532).

The top-level client call should then decide whether it wants to be tracing or not. I'll do a follow-up PR to address this after this and #16157 land.